lurk-verify/src/lib.rs

```diff
@@ -0,0 +1,224 @@
+use lurk::{
+    field::LurkField,
+    proof::nova::PublicParams,
+    store::{ScalarPointer, ScalarPtr, Store},
+};
+use pasta_curves::pallas;
+use serde::{Deserialize, Serialize};
+use std::convert::TryFrom;
+use std::fs::File;
+use std::io::{self, BufReader, BufWriter};
+use std::path::Path;
+
+pub use crate::error::VerifyError;
+
+pub mod error;
+
+pub const DEFAULT_REDUCTION_COUNT: ReductionCount = ReductionCount::One;
+
+pub type S1 = pallas::Scalar;
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct VerificationResult {
+    pub verified: bool,
+}
+
+impl VerificationResult {
+    fn new(verified: bool) -> Self {
+        Self { verified }
+    }
+}
+
+// Number of circuit reductions per step, equivalent to `chunk_frame_count`
+#[derive(Clone, Copy, Hash, PartialEq, Eq, Serialize, Deserialize)]
+pub enum ReductionCount {
+    One,
+    Five,
+    Ten,
+}
+
+impl TryFrom<usize> for ReductionCount {
+    type Error = String;
+
+    fn try_from(count: usize) -> Result<Self, String> {
+        match count {
+            1 => Ok(ReductionCount::One),
+            5 => Ok(ReductionCount::Five),
+            10 => Ok(ReductionCount::Ten),
+            c => Err(format!("Unsupported Reduction Count: {c}")),
+        }
+    }
+}
+impl ReductionCount {
+    pub fn count(&self) -> usize {
+        match self {
+            Self::One => 1,
+            Self::Five => 5,
+            Self::Ten => 10,
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct LurkProof<'a, F: LurkField> {
+    #[serde(bound(
+        serialize = "Claim<F>: Serialize",
+        deserialize = "Claim<F>: Deserialize<'de>"
+    ))]
+    pub claim: Option<Claim<F>>,
+    pub public_inputs: Vec<S1>,
+    pub public_outputs: Vec<S1>,
```

Comment:
Why require `public_inputs` and `public_outputs` if the claim can be supplied? This just makes the caller produce the I/O themselves (from the claim).

If for some reason someone supplies everything, you can still check for consistency as below, but my assumption is that use cases where claim and I/O are available are largely disjoint. So I would support usage that supplies only one or other.

Unless you really want to support the case where inconsistent claim and I/O are possible (and need to be checked), it make sense to define a new enum (maybe `Statement`) that can be either `Claim` or `PublicIO`, and then let `PublicIO` have an `input` and `output` field.